homeassistant/components/integration/sensor.py

```diff
@@ -143,7 +161,11 @@ def calc_integration(event):
                 self._unit_of_measurement = self._unit_template.format(
                     "" if unit is None else unit
                 )
-
+            if (
+                self.device_class is None
+                and new_state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_POWER
+            ):
+                self._attr_device_class = DEVICE_CLASS_ENERGY
```

Comment:
we should be able to restore device_class together with last_reset

Reply:
this is determined by the device class upstream? Why should it be stored/restored?

Reply:
it might not be set until you get the first state update ?

Reply:
True, added a suggestion above that implements that.